src/modules/job-ingest/validate.c

```diff
@@ -83,25 +84,45 @@ static bool str_ends_with (const char *str, const char *suffix)
 
 struct validate *validate_create (flux_t *h,
                                   const char *validate_path,
-                                  const char *schema_path)
+                                  const char *validator_args)
 {
     struct validate *v;
     char *argv[5];
     int argc = 0;
     int i;
+    char *validator_argz = NULL;
+    char *validator_arg = NULL;
+    size_t validator_argz_len = 0;
 
     if (!(v = calloc (1, sizeof (*v))))
         return NULL;
     v->h = h;
 
     assert (validate_path != NULL);
-    assert (schema_path != NULL);
 
     if (str_ends_with (validate_path, ".py"))
         argv[argc++] = PYTHON_INTERPRETER;
     argv[argc++] = (char *)validate_path;
-    argv[argc++] = "--schema";
-    argv[argc++] = (char *)schema_path,
+    if (validator_args != NULL) {
+        // Parse the comma-separated argument list passed in when loading the
+        // job-ingest module.  For example:
+        // module load job-ingest validator-args=--schema,/path/to/schema.json
+        if (argz_create_sep (validator_args,
```

Comment:
does there need to be a corresponding `free()` call somewhere to avoid a leak when the module is unloaded?

Reply:
Good catch! Just pushed a fix.